code/__LANGUAGE_CONSTRUCTS/getters_setters.dm

```diff
@@ -0,0 +1,151 @@
+
+/*
+SHIT'S MAGIC, By Remie Richards
+
+HOW TO USE:
+
+PROP:
+	var/somevar = someval
+	IS NOW
+	PROP(somevar, someval)
+
+	The auto generated variables are:
+	prop_somevar
+	prop_getters_somevar
+	prop_setters_somevar
+
+	However you should only access "somevar" (without the quotes), using the correct macros
+	Using these macros will convert somevar into prop_somevar in the background naturally
+	Intended to make this whole thing feel nicer to use
+
+PROPN:
+	Same as above except the getter/setter lists start null
+	This lets you avoid byond's hidden, costly Init() procs
+
+LAZYINIT_PROPLISTS:
+	Initialises all non-null getter/setter lists to an empty list
+	Useful only if you *NEED* all getter/setter lists to exist and you used PROPN()
+
+GET:
+	var/variable = someobj.somevar
+	IS NOW
+	var/variable = GET(someobj, somevar)
+
+SET:
+	someobj.somevar = someval
+	IS NOW
+	SET(someobj, somevar, someval)
+
+[SET/GET]_MODIFY:
+	Same as their above equivalents, except they have slightly higher costs due to allowing for the returned/set value to be modified
+	GET_MODIFY is more costly (only one proc call more) than SET_MODIFY due to needing to call a proc for ease-of-use
+
+	*The non-MODIFY versions can be considered read/write traps, as opposed to getters/setters
+
+ADD_[SETTER/GETTER]:
+	Wrappers around LAZYADD() that add a setter/getter callback datum to the appropriate list
+
+REMOVE_[SETTER/GETTER]:
+	Wrappers around LAZYREMOVE() that remove a setter/getter callback datum from the appropriate list
+
+*/
+
+
+//Define a propery + associated set/get lists
+#define PROP(propname, propval) 				\
+	var/prop_##propname = ##propval;			\
```

Comment:
I considered it, and even made a version that allows it, however I quickly realised since you're never directly accessing the variable (always through GET()) you have to put it in a new var, which can be typed as you wish already.

(You could type the actual variable itself, prop_propname, but those vars aren't intended for direct access, so I wish to discourage that, typing them lends unwanted legitimacy to that way of accessing)